src/main/java/com/sparta/trillionnewspeedproject/dto/PostResponseDto.java

```diff
@@ -0,0 +1,37 @@
+package com.sparta.trillionnewspeedproject.dto;
+
+import com.sparta.trillionnewspeedproject.entity.Post;
+import com.sparta.trillionnewspeedproject.dto.ApiResponseDto;
+import com.sparta.trillionnewspeedproject.dto.CommentResponseDto;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.time.LocalDateTime;
+import java.util.Comparator;
+import java.util.List;
+
+@Getter
+@Setter
+public class PostResponseDto extends ApiResponseDto {
+    private Long id;
+    private String title;
+    private String content;
+    private LocalDateTime createdAt;
+    private LocalDateTime modifiedAt;
+    private String username;
+    private List<CommentResponseDto> comments;
+
+    public PostResponseDto(Post post) {
+        this.id = post.getPost_id();
+        this.title = post.getTitle();
+        this.content = post.getContent();
+        this.createdAt = post.getCreatedAt();
+        this.modifiedAt = post.getModifiedAt();
+        this.username = post.getUser().getUsername();
+        this.comments = post.getComments().stream()
+            .map(CommentResponseDto::new)
+            .sorted(Comparator.comparing(CommentResponseDto::getCreatedAt).reversed()) // 작성날짜 내림차순 - reversed,
+                // getCreatedAt - 작성일자, comparing - 비교 연산자, sorted - 정렬
+            .toList();
```

Comment:
수정이나 조회에서는 별문제 없는데 작성 시에 

java.lang.NullPointerException: Cannot invoke "java.util.List.stream()" because the return value of "com.sparta.trillionnewspeedproject.entity.Post.getComments()" is null

이런 Exception이 생겨서 DB에 저장은 되는데 Reponse에 반환이 안되네요. 혹시 제가 또 뭘 놓친걸까요..?ㅠㅠ

Reply:
저도 해당 부분 보고 있습니다... 반환시에 댓글을 바로 반화해서 현상 발생하는 거 같은데 어떻게 처리 해야할지..

Reply:
저는 이전에 과제할 때 postResponseDto에는 comment를 제외하고 service단에서 구현하긴했어요. 좋은 코드는 아니지만 이런 코드도 있구나 참고가 될까하여 올려봅니다.

`// 전체 게시글 및 댓글 목록 조회
    public List<List<Object>> getPosts() {
        List<Post> postList = postRepository.findAllByOrderByCreatedAtDesc();

        List<List<Object>> postAndCommentsList = new ArrayList<>();

        for (int i = 0; i < postList.size(); i++) {
            postAndCommentsList.add(getPostById(postList.get(i).getId()));
        }
        return postAndCommentsList;
    }

    // 선택한 게시글 및 댓글 조회
    @Transactional
    public List<Object> getPostById(Long id) {
        List<Object> postAndComments = new ArrayList<>();
        postAndComments.add(new PostResponseDto(findPost(id)));
        postAndComments.add(commentService.getCommentsByPostId(id));
        return postAndComments;
    }`

Reply:
public PostResponseDto(Post post) {
        this.id = post.getPost_id();
        this.title = post.getTitle();
        this.content = post.getContent();
        this.createdAt = post.getCreatedAt();
        this.modifiedAt = post.getModifiedAt();
        this.username = post.getUser().getUsername();
        if (!(post.getComments() == null)) {
            this.comments = post.getComments().stream()
                    .map(CommentResponseDto::new)
                    .sorted(Comparator.comparing(CommentResponseDto::getCreatedAt).reversed()) // 작성날짜 내림차순 - reversed,
                    // getCreatedAt - 작성일자, comparing - 비교 연산자, sorted - 정렬
                    .toList();
        }

Reply:
오오 아직 테스트로 확인은 못해보긴 했는데 작성할 때 오류가 없으면 괜찮을 것 같습니다! 생각해보니 NullPointerExeption이니까 null만 잡으면 되는데 저렇게 if문을 쓸 생각을 못했었네요!! 고생 많으셨습니다!!👍👍